GoogleCloudPlatform/bank-of-anthos
# fix: catch decimal error and return fail

### Fixes #182

### Background 
- Whenever an invalid decimal number is input the UI does not allow the payment to be submitted. However, if someone directly calls the API then the server just fails with an exception
- They primarily fail because the invalid decimal number raises an exception which is not caught resulting in no response from server

### Additional Notes
- After investigating further, we notice that the real reason [for the specific cypress test](https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/master/.github/workflows/ui-tests/cypress/integration/transfer_spec.js#L194-L202) to fail _(in the current codebase)_ is because that the [input field for the amount](https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/master/src/frontend/templates/index.html#L331) has been set to **number=type**. This means there is no way a user can input an invalid number into that field because the browser will not allow it. The same happens when cypress sets an invalid value to this UI element, it automatically rounds to the closest valid decimal. Thus the transaction goes through without an issue - resulting in the test to fail. 
- So this PR removes the test completely - because that doesn't work anymore; however since someone could still submit an invalid number by calling the API directly - we catch this exception and return an error. 
- Some of the CI files have been update to match the test pr #545, in order to see if the cypress tests pass
- These changes must be reverted before merge

### Change Summary
- Catch the exception raised by the invalid number input
- Return an error message
- Remove the Cypress test that checks the UI element


### Testing Procedure
- Cypress tests should pass + not have the pending test for [the removed test](https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/master/.github/workflows/ui-tests/cypress/integration/transfer_spec.js#L194-L202).
- You can try sending a valid request to the server, then copying the request as `Curl` from the Chrome inspect window
- Resubmit the curl with an invalid value from your terminal

### Related PRs or Issues 
- N/A
